crates/yamlpatch/src/lib.rs

```diff
@@ -209,6 +209,43 @@ pub fn apply_yaml_patches(
     Ok(next_document)
 }
 
+/// Compute the replacement range for a byte span, including any trailing newline
+/// that exists immediately after the span in the document.
+///
+/// This fixes the root cause: tree-sitter's byte span doesn't include trailing
+/// newlines that come after the feature node, so we need to extend the range
+/// to include them to avoid losing them during replacement.
+///
+/// We only include the trailing newline if it's the last character in the document,
+/// to avoid including newlines that are part of the document structure rather than
+/// a document-level trailing newline.
+fn compute_replacement_range_for_span(
+    byte_span: (usize, usize),
+    document: &yamlpath::Document,
+) -> std::ops::Range<usize> {
+    let start = byte_span.0;
+    let mut end = byte_span.1;
+    let content = document.source();
+
+    // Only include trailing newline if it's the last character in the document.
+    // This preserves document-level trailing newlines without interfering with
+    // newlines that are part of the document structure.
+    if end < content.len() && content.as_bytes()[end] == b'\n' && end + 1 == content.len() {
```

Comment:
I might be underthinking this, but I think this is shorter and equivalent:

```suggestion
    if content.as_bytes()[end] == b'\n' && end + 1 == content.len() {
```

Since `end + 1 == content.len()` implies `env < content.len()` 🙂